Kerdy/Kerdy/InitialSetting/VC/ViewModel/CategoryViewModel.swift

```diff
@@ -0,0 +1,30 @@
+//
+//  CategoryViewModel.swift
+//  Kerdy
+//
+//  Created by 최다경 on 11/5/23.
+//
+
+import UIKit
+import RxSwift
+
+class CategoryViewModel {
+    let selectCount = BehaviorSubject<Int>(value: 0)
+    
+    var categorySelectedDict: [UIButton: BehaviorSubject<Bool>] = [:]
+    
+    func categoryButtonTapped(button: UIButton) {
+        if let buttonState = try? categorySelectedDict[button]?.value() {
+            if buttonState == true {
+                categorySelectedDict[button]?.onNext(false)
+                if let count = try? selectCount.value() {
+                    selectCount.onNext(count - 1)
+                }
+            } else if buttonState == false, let count = try? selectCount.value(), count < 4 {
+                categorySelectedDict[button]?.onNext(true)
+                selectCount.onNext(count + 1)
+            }
+        }
+    }
```

Comment:
if 문 중첩이 많을 경우 코드의 가독성이 떨어집니다. 아래와 같이 guard 문을 사용해보면 어떨까요?
```suggestion
func categoryButtonTapped(button: UIButton) {
    guard let buttonState = try? categorySelectedDict[button]?.value(),
          let count = try? selectCount.value() else {
        // categorySelectedDict 또는 selectCount.value가 nil일 경우 처리
        return
    }
    
    if buttonState == true {
        categorySelectedDict[button]?.onNext(false)
        selectCount.onNext(count - 1)
    } else if buttonState == false, count < 4 {
        categorySelectedDict[button]?.onNext(true)
        selectCount.onNext(count + 1)
    }
}
```

Reply:
수정했습니다 😃